aiomisc/service/aiohttp.py

```diff
@@ -0,0 +1,58 @@
+import socket
+
+from aiohttp.web import Application, AppRunner, SockSite
+from aiohttp.helpers import AccessLogger
+
+from aiomisc.utils import bind_socket
+from .base import Service
+
+
+class AIOHTTPService(Service):
+    __async_required__ = frozenset({'start', 'create_application'})
+
+    def __init__(self, address: str = None, port: int = None,
+                 sock=None, shutdown_timeout: int = 5):
+
+        if sock is not None:
+            self.socket = sock
+        elif address and port:
+            self.socket = bind_socket(
+                address=address,
+                port=port,
+                proto_name='http',
+            )
+        elif not isinstance(sock, socket.socket):
```

Comment:
This branch will be executed if `sock`, `address` and `port` are all three is `None`, but logically branch below should be executed instead.